extensions/aligning_guidelines/index.ts

```diff
@@ -32,113 +28,153 @@ export type { AligningLineConfig } from './typedefs';
+    target.setCoords();
     onlyDrawPoint = false;
     verticalLines.clear();
     horizontalLines.clear();
 
-    const objects = getObjectsByTarget(activeObject);
-    const activeObjectRect = activeObject.getBoundingRect();
-
-    for (const object of objects) {
-      const objectRect = getCaCheMapValue(object)[0];
-      const { vLines, hLines } = collectLine({
-        activeObject,
-        activeObjectRect,
-        objectRect,
-      });
-      vLines.forEach((o) => {
-        verticalLines.add(JSON.stringify(o));
-      });
-      hLines.forEach((o) => {
-        horizontalLines.add(JSON.stringify(o));
-      });
-    }
+    // Find the shapes associated with the current graphic to draw reference lines for it.
+    const objects =
+      options.getObjectsByTarget?.(target) ?? getObjectsByTarget(target);
+    const points: Point[] = [];
+    // Collect all the points to draw reference lines.
+    for (const object of objects) points.push(...getCaCheMapValue(object));
+
+    // Obtain horizontal and vertical reference lines.
+    const { vLines, hLines } = collectLine(target, points);
+    vLines.forEach((o) => {
+      // Objects cannot be deduplicated; convert them to strings for deduplication.
+      verticalLines.add(JSON.stringify(o));
+    });
+    hLines.forEach((o) => {
+      // Objects cannot be deduplicated; convert them to strings for deduplication.
+      horizontalLines.add(JSON.stringify(o));
+    });
   }
 
   function scalingOrResizing(e: TransformEvent) {
-    // br bl tr tl mb ml mt mr
-    const activeObject = e.target;
-    activeObject.setCoords();
+    const target = e.target;
+    // We need to obtain the real-time coordinates of the current object, so we need to update them in real-time
+    target.setCoords();
+    // The value of action can be scaleX, scaleY, scale, resize, etc.
+    // If it does not start with "scale," it is considered a modification of size.
     const isScale = String(e.transform.action).startsWith('scale');
     verticalLines.clear();
     horizontalLines.clear();
 
-    const objects = getObjectsByTarget(activeObject);
+    const objects =
+      options.getObjectsByTarget?.(target) ?? getObjectsByTarget(target);
     let corner = e.transform.corner;
-    if (activeObject.flipX) corner = corner.replace('l', 'r').replace('r', 'l');
-    if (activeObject.flipY) corner = corner.replace('t', 'b').replace('b', 't');
-    let index = ['tl', 'tr', 'br', 'bl', 'mt', 'mr', 'mb', 'ml'].indexOf(
-      corner,
-    );
-    if (index == -1) return;
-    onlyDrawPoint = index > 3;
+    // When the shape is flipped, the tl obtained through getCoords is actually tr,
+    // and tl is actually tr. We need to make correction adjustments.
+    // tr <-> tl、 bl <-> br、  mb <-> mt、 ml <-> mr
+    if (target.flipX) corner = corner.replace('l', 'r').replace('r', 'l');
+    if (target.flipY) corner = corner.replace('t', 'b').replace('b', 't');
+    const coords = target.getCoords();
```

Comment:
I can't see properly but is this used?

Reply:
When executing the flip, its value becomes true, which is set internally by Fabric. The video below shows the bug that occurs when it is turned off.


https://github.com/user-attachments/assets/71802194-b2b5-4f34-81e0-737bf8ae85a9

Reply:
I mean that `const coords = target.getCoords();` seems unused

Reply:
Yes, you are right, it should be deleted.